pkg/apis/rook.io/v1alpha2/network.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1alpha2
+
+import (
+	"encoding/json"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func IsHostNetworkDefined(net NetworkSpec) bool {
+	return net.Provider == "host"
+}
+
+func IsMultusNetworkDefined(net NetworkSpec) bool {
+	return !IsHostNetworkDefined(net) && net.Provider == "multus"
+}
+
+func parseMultusSelector(ns NetworkSelector) map[string]string {
+	rawMap := make(map[string]string)
+
+	err := json.Unmarshal([]byte(ns), &rawMap)
+
+	if err != nil {
+		// it can be in short form
+		multusSelectorString := string(ns)
+
+		nsEndIndex := strings.IndexAny(multusSelectorString, "/")
```

Comment:
Could you add a comment with an example of what we're parsing?

Reply:
Ok, I will use docs style comment.